programmers-coding-test/minjae/dfs-bfs/미로탈출.js

```diff
@@ -0,0 +1,44 @@
+const maze = [
+    [1, 0, 1, 0, 1, 0],
+    [1, 1, 1, 1, 1, 1],
+    [0, 0, 0, 0, 0, 1],
+    [1, 1, 1, 1, 1, 1],
+    [1, 1, 1, 1, 1, 1],
+];
+
+const MAZE_X = maze.length;
+const MAZE_Y = maze[0].length;
+
+//BFS는 큐2개 필요
+function solution(x, y) {
+    const dx = [-1, 1, 0, 0];
+    const dy = [0, 0, -1, 1];
+
+    let visitedQueue, needVisitQueue; 
+    visitedQueue = needVisitQueue = [];
+
+    let nx, ny;
+    nx = ny = 0;
+
+    needVisitQueue.push([x, y]);
+    
+    while (needVisitQueue.length !== 0) {
+        const node = needVisitQueue.shift();
+        [x, y] = node;
+        
+        for (let i = 0; i < 4; i++) {
+            nx = x + dx[i];
+            ny = y + dy[i];
+
+            if (nx < 0 || ny < 0 || nx >= MAZE_X || ny >= MAZE_Y) continue;
+            else if (maze[nx][ny] === 0) continue;
+            else if (maze[nx][ny] === 1) {
```

Comment:
continue 때문에 해당 조건이 일치하면 아래 구문은 실행하지 않습니다.
else if 를 사용하지 않아도 되는거죠!ㅎㅎ
```suggestion
            if (maze[nx][ny] === 0) continue;
            if (maze[nx][ny] === 1) {
```